packages/lerna-semantic-release-pre/index.js

```diff
@@ -89,26 +99,87 @@ function bumpVersionCommitAndTag (nextRelease, done) {
   });
 }
 
+function updateDependencies (pkg, updates, dependenciesKey) {
+  var deps = pkg[dependenciesKey];
+
+  if (!deps) {
+    return pkg;
+  }
+
+  log.info('Updating ' + dependenciesKey + ' of ' + pkg.name);
+
+  pkg[dependenciesKey] = Object.keys(deps).reduce(function (dependenciesObj, depName) {
+    var update = updates[depName];
+
+    if (!update) {
+      // keep dependency version the same
+      dependenciesObj[depName] = deps[depName];
+      return dependenciesObj;
+    }
+
+    var oldVersion = deps[depName];
+    var newVersion = '^' + update.nextRelease.version;
+
+    log.info(depName + ' is updated: ' + oldVersion + ' -> ' + newVersion);
+
+    dependenciesObj[depName] = newVersion;
+    return dependenciesObj;
+  }, {});
+
+  return pkg;
+}
+
+function savePkg (pkg, packagePath, fs) {
+  return function (done) {
+    fs.writeFile(getPkgLocation(packagePath), JSON.stringify(pkg, null, 2), done);
+  }
+}
+
 module.exports = function (config) {
   config.io.git.head(function (err, releaseHash) {
     err && log.error(err);
 
     forEachPackage([
       getNpmConfig,
       makeSrConfig,
-      pre,
-      bumpVersionCommitAndTag
+      pre
     ], {
       allPackages: config.io.lerna.getAllPackages(),
       asyncType: async.waterfall,
       extraContext:  {
         releaseHash: releaseHash,
         io: config.io
       }
-    }, (err) => {
-      if (typeof config.callback === 'function') {
+    }, (err, results) => {
+      if (err && typeof config.callback === 'function') {
         config.callback(err);
+        return;
       }
+
+      var updates = results.reduce(function (acc, nameAndUpdatePair) {
+        var name = nameAndUpdatePair[0];
+        var update = nameAndUpdatePair[1];
+
+        if (!update.nextRelease) {
+          // skip packages that are not going to be released
+          return acc;
+        }
+
+        acc[name] = update;
+        return acc;
+      }, {});
+
+      var tasks = Object.keys(updates).map(function (updatedPackageName) {
+        return function(done) {
+          bumpVersionCommitAndTag(updatedPackageName, updates, releaseHash, config.io, done);
+        }
+      });
```

Comment:
Would it be possible to move this into a new function with a descriptive name? I think it'd help to make this section short (eg. `updateToNewDependencies`)